File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.java
class Solution {
  public int nearestValidPoint(int x, int y, int[][] points) {
    int result = -1;
    int min = Integer.MAX_VALUE;

    for (int i = 0; i < points.length; ++i) {
      final int dx = x - points[i][0];
      final int dy = y - points[i][1];
      if (dx == 0 || dy == 0) {
        final int distance = Math.abs(dx + dy);
        if (distance < min) {
          min = distance;
          result = i;
        }
      }
    }

    return result;
  }
}

File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.py
class Solution:
   def nearestValidPoint(self, x, y, points):
        init_d=float('inf')
        result=-1
        for i in range(len(points)):
            
            x0 = points[i][0]
            y0 = points[i][1]
            dx = x - x0 
            dy = y - y0
            
            if dx == 0 or dy == 0:
                distance = abs(dx) + abs(dy)
                if distance < init_d:
                    init_d, result = distance, i
        return result
